### ISSUE TITLE
beaker.experiment.list(..., finalized=True) hangs

### ISSUE THREAD

**Issue by TimDettmers** [allenai/beaker-py#283]
### 🐛 Describe the bug

I want to get past experiments that I ran and I use
```python
from beaker import Beaker
beaker = Beaker.from_env(default_workspace="ai2/timd")
print(beaker.experiment.list(author='timd', finalized=True))
```
The following request is constructed in `service_client.py` line 89:
```
url: https://beaker.org/api/v3/jobs?author=01J47XXAYGSQT651Q6XHNP55HW&kind=execution&finalized=True
headers: {'Authorization': '<token>', 'Content-Type': 'application/json', 'User-Agent': 'beaker-py v1.31.2'}
request_data: None
```

The response hangs and never finishes.

### Versions

Python 3.10.9
absl-py==1.4.0
accelerate==0.32.1
aiofiles==23.1.0
aiohttp==3.8.1
aiosignal==1.3.1
alembic==1.10.4
altair==5.0.0
antlr4-python3-runtime==4.9.3
anyio==3.6.2
appdirs==1.4.4
APScheduler==3.10.1
arrow==1.2.3
astunparse==1.6.3
async-timeout==4.0.2
attrs==22.2.0
audioread==3.0.0
av==9.2.0
awscli==1.29.55
Babel==2.12.1
backoff==1.11.1
beaker-gantry==1.8.3
beaker-py==1.31.2
beautifulsoup4==4.12.2
bert-score==0.3.13
binaryornot==0.4.4
-e git+ssh://git@github.com/bitsandbytes-foundation/bitsandbytes.git@91ea416f3b61963a173638c65308b4371fd80916#egg=bitsandbytes
black==23.3.0
bleach==6.0.0
blinker==1.6.2
boltons @ file:///croot/boltons_1677628692245/work
botocore==1.31.55
brotlipy==0.7.0
build==0.10.0
cached-property==1.5.2
cachetools==5.3.0
certifi @ file:///croot/certifi_1671487769961/work/certifi
cffi @ file:///croot/cffi_1670423208954/work
chardet==5.1.0
charset-normalizer @ file:///tmp/build/80754af9/charset-normalizer_1630003229654/work
chex==0.1.6
click==8.1.3
click-help-colors==0.9.4
clldutils==3.19.0
cmaes==0.9.1
cmake==3.26.3
codecarbon==1.2.0
colorama==0.4.4
colorlog==6.7.0
commonmark==0.9.1
conda==23.3.1
conda-content-trust @ file:///tmp/abs_5952f1c8-355c-4855-ad2e-538535021ba5h26t22e5/croots/recipe/conda-content-trust_1658126371814/work
conda-package-handling @ file:///croot/conda-package-handling_1672865015732/work
conda_package_streaming @ file:///croot/conda-package-streaming_1670508151586/work
contourpy==1.0.7
cookiecutter==1.7.3
coolname==2.2.0
cryptography @ file:///croot/cryptography_1677533068310/work
csvw==3.1.3
cycler==0.11.0
dash==2.9.3
dash-bootstrap-components==1.4.1
dash-core-components==2.0.0
dash-html-components==2.0.0
dash-table==5.0.0
DataProperty==1.0.1
datasets==2.20.0
decorator==5.1.1
decord==0.6.0
Deprecated==1.2.13
dill==0.3.4
distlib==0.3.6
dlinfo==1.2.1
dm-tree==0.1.8
docker==7.1.0
docker-pycreds==0.4.0
docstring_parser==0.16
docutils==0.16
einops==0.8.0
etils==1.2.0
evaluate==0.4.0
exceptiongroup==1.1.1
execnet==1.9.0
faiss-cpu==1.7.4
fastapi==0.95.2
fastjsonschema==2.16.3
ffmpy==0.3.0
filelock @ file:///croot/filelock_1672387128942/work
fire==0.5.0
Flask==2.3.2
flatbuffers==24.3.25
flax==0.6.9
fonttools==4.39.3
frozenlist==1.3.3
fsspec==2023.5.0
fugashi==1.2.1
gast==0.4.0
gekko==1.0.6
gitdb==4.0.10
GitPython==3.1.18
gmpy2 @ file:///tmp/build/80754af9/gmpy2_1645455533097/work
google-auth==2.17.3
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
gql==3.4.1
gradio==3.32.0
gradio_client==0.2.5
graphql-core==3.2.3
greenlet==2.0.2
grpcio==1.51.3
h11==0.14.0
h5py==3.11.0
hf-doc-builder==0.4.0
httpcore==0.17.2
httplib2==0.22.0
httpx==0.24.1
huggingface-hub==0.23.4
hypothesis==6.75.2
idna @ file:///croot/idna_1666125576474/work
importlib-metadata==6.3.0
importlib-resources==5.12.0
iniconfig==2.0.0
ipadic==1.0.0
isodate==0.6.1
isort==5.12.0
itsdangerous==2.1.2
jaraco.classes==3.2.3
jax==0.4.13
jaxlib==0.4.13
jeepney==0.8.0
Jinja2 @ file:///croot/jinja2_1666908132255/work
jinja2-time==0.2.0
jmespath==1.0.1
joblib==1.2.0
jsonlines==4.0.0
jsonpatch @ file:///tmp/build/80754af9/jsonpatch_1615747632069/work
jsonpointer==2.1
jsonschema==4.17.3
jupyter_core==5.3.0
kenlm==0.1
keras==3.4.1
keras-nlp==0.5.1
keyring==23.13.1
kiwisolver==1.4.4
language-tags==1.2.0
lazy_loader==0.2
libclang==16.0.0
librosa==0.10.0.post2
lightning-utilities==0.9.0
linkify-it-py==2.0.2
lion-pytorch==0.2.2
lit==16.0.1
llvmlite==0.39.1
lxml==4.9.2
Mako==1.2.4
Markdown==3.4.3
markdown-it-py==2.2.0
MarkupSafe @ file:///opt/conda/conda-bld/markupsafe_1654597864307/work
matplotlib==3.9.1
mbstrdecoder==1.1.3
mdit-py-plugins==0.3.3
mdurl==0.1.2
mkl-fft==1.3.1
mkl-random @ file:///home/builder/ci_310/mkl_random_1641843545607/work
mkl-service==2.4.0
ml-dtypes==0.4.0
mmm==0.4.2
more-itertools==9.1.0
mpmath==1.2.1
msgpack==1.0.5
multidict==6.0.4
multiprocess==0.70.12.2
mypy-extensions==1.0.0
namex==0.0.8
nbformat==5.8.0
nest-asyncio==1.5.6
networkx @ file:///croot/networkx_1678964333703/work
nltk==3.8.1
numexpr==2.8.5
numpy==1.26.4
nvidia-cublas-cu12==12.1.3.1
nvidia-cuda-cupti-cu12==12.1.105
nvidia-cuda-nvrtc-cu12==12.1.105
nvidia-cuda-runtime-cu12==12.1.105
nvidia-cudnn-cu12==8.9.2.26
nvidia-cufft-cu12==11.0.2.54
nvidia-curand-cu12==10.3.2.106
nvidia-cusolver-cu12==11.4.5.107
nvidia-cusparse-cu12==12.1.0.106
nvidia-nccl-cu12==2.20.5
nvidia-nvjitlink-cu12==12.5.82
nvidia-nvtx-cu12==12.1.105
oauthlib==3.2.2
omegaconf==2.4.0.dev0
onnxconverter-common==1.13.0
openai==0.28.0
opt-einsum==3.3.0
optax==0.1.4
optree==0.12.1
optuna==3.1.1
orbax-checkpoint==0.2.1
orjson==3.8.13
packaging @ file:///croot/packaging_1678965309396/work
pandas==2.2.2
parameterized==0.9.0
pathspec==0.11.1
pathtools==0.1.2
pathvalidate==3.1.0
peft==0.11.1
petname==2.6
phonemizer==3.2.1
Pillow==9.4.0
pkginfo==1.9.6
plac==1.3.5
platformdirs==3.5.0
plotly==5.14.1
pluggy==1.5.0
pooch==1.6.0
portalocker==2.0.0
poyo==0.5.0
protobuf==4.25.3
psutil==5.9.5
py-cpuinfo==9.0.0
pyarrow==16.1.0
pyarrow-hotfix==0.6
pyasn1==0.5.0
pyasn1-modules==0.3.0
pybind11==2.11.1
pycosat @ file:///croot/pycosat_1666805502580/work
pycountry==22.3.5
pycparser @ file:///tmp/build/80754af9/pycparser_1636541352034/work
pyctcdecode==0.5.0
pydantic==1.10.8
pydub==0.25.1
PyGithub==1.58.1
Pygments==2.15.0
pygtrie==2.5.0
PyJWT==2.6.0
pylatexenc==2.10
PyNaCl==1.5.0
pynvml==11.5.0
pyOpenSSL @ file:///croot/pyopenssl_1677607685877/work
pyparsing==3.0.9
pypng==0.20220715.0
pyproject_hooks==1.0.0
pyre-extensions==0.0.29
pyrsistent==0.19.3
PySocks @ file:///home/builder/ci_310/pysocks_1640793678128/work
pytablewriter==1.0.0
pytest==8.2.2
pytest-timeout==2.1.0
pytest-xdist==3.2.1
python-dateutil==2.8.2
python-multipart==0.0.6
python-slugify==8.0.1
pytz==2023.3
pytz-deprecation-shim==0.1.0.post0
PyYAML==5.4.1
ray==2.4.0
rdflib==6.3.2
readme-renderer==37.3
regex==2023.3.23
requests==2.32.3
requests-oauthlib==1.3.1
requests-toolbelt==0.10.1
responses==0.18.0
rfc3986==1.5.0
rhoknp==1.3.0
rich==13.3.5
rjieba==0.1.11
rouge-score==0.1.2
rsa==4.7.2
ruamel.yaml @ file:///croot/ruamel.yaml_1666304550667/work
ruamel.yaml.clib @ file:///croot/ruamel.yaml.clib_1666302247304/work
ruff==0.0.259
s3transfer==0.7.0
sacrebleu==1.5.1
sacremoses==0.0.53
safetensors==0.4.3
scikit-learn==1.2.2
scipy==1.14.0
seaborn==0.12.2
SecretStorage==3.3.3
segments==2.2.1
semantic-version==2.10.0
sentencepiece==0.1.99
sentry-sdk==1.22.2
setproctitle==1.3.2
shtab==1.7.1
sigopt==8.8.0
six @ file:///tmp/build/80754af9/six_1644875935023/work
smmap==5.0.0
sniffio==1.3.0
sortedcontainers==2.4.0
soundfile==0.12.1
soupsieve==2.4.1
soxr==0.3.5
SQLAlchemy==2.0.12
sqlitedict==2.1.0
starlette==0.27.0
SudachiDict-core==20230110
SudachiPy==0.6.7
sympy==1.12
tabledata==1.3.1
tabulate==0.9.0
tcolorpy==0.1.3
tenacity==8.2.2
tensorboard==2.17.0
tensorboard-data-server==0.7.2
tensorboard-plugin-wit==1.8.1
tensorflow==2.17.0
tensorflow-estimator==2.11.0
tensorflow-hub==0.13.0
tensorflow-io-gcs-filesystem==0.32.0
tensorflow-text==2.11.0
tensorstore==0.1.36
termcolor==2.3.0
text-unidecode==1.3
tf_keras==2.17.0
threadpoolctl==3.1.0
timeout-decorator==0.5.0
timm==0.6.13
tokenizers==0.19.1
tomli==2.0.1
toolz @ file:///croot/toolz_1667464077321/work
torch==2.3.1
torchaudio==2.0.0
torchmetrics==1.0.0
torchvision==0.15.0
tqdm==4.66.4
tqdm-multiprocess==0.0.11
traitlets==5.9.0
transformers==4.42.4
triton==2.3.1
trl==0.9.6
twine==4.0.2
typepy==1.3.1
typing-inspect==0.8.0
typing_extensions==4.12.2
tyro==0.8.5
tzdata==2023.3
tzlocal==4.3
uc-micro-py==1.0.2
unidic==1.1.0
unidic-lite==1.0.8
uritemplate==4.1.1
urllib3 @ file:///croot/urllib3_1680254681959/work
uvicorn==0.22.0
virtualenv==20.21.0
wandb==0.15.2
wasabi==0.10.1
webencodings==0.5.1
websockets==11.0.3
Werkzeug==2.3.3
wrapt==1.15.0
xxhash==3.2.0
yarl==1.8.2
zipp==3.15.0
zstandard @ file:///croot/zstandard_1677013143055/work

**Comment by epwalsh** [comment]
@mewil any ideas? If I change `finalized` to `False` I get a quick response.

**Comment by TimDettmers** [comment]
A workaround is this code from @OyvindTafjord:

```python
from beaker import Beaker
BEAKER = Beaker.from_env()
exps = BEAKER.workspace.experiments("ai2/timd")
```

which works well for me. So, practically, this solves my issue. Not sure what the difference would be between those two methods.

**Comment by stupendousC** [comment]
https://beaker.org/api/v3/jobs?author=01J47XXAYGSQT651Q6XHNP55HW&kind=execution&finalized=True on my browser took >10sec to kick in, so I was thinking... 

What if you jus increase the timeout to 30 seconds or something?  
https://github.com/allenai/beaker-py/blob/df27ac8e75395dc9613318b9c7c92a4b11d48a90/beaker/client.py#L82
